x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/ip/IpPrefixTests.java

```diff
@@ -0,0 +1,96 @@
+            // V4
+            new TestCaseSupplier(
+                List.of(DataTypes.IP, DataTypes.INTEGER),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("1.2.3.4"), DataTypes.IP, "ip"),
+                        new TestCaseSupplier.TypedData(24, DataTypes.INTEGER, "prefixLengthV4"),
+                        new TestCaseSupplier.TypedData(127, DataTypes.INTEGER, "prefixLengthV6")
+                    ),
+                    "IpPrefixEvaluator[ip=Attribute[channel=0], prefixLengthV4=Attribute[channel=1], prefixLengthV6=Attribute[channel=2]]",
+                    DataTypes.IP,
+                    equalTo(EsqlDataTypeConverter.stringToIP("1.2.3.0"))
+                )
+            ),
+            new TestCaseSupplier(
+                List.of(DataTypes.IP, DataTypes.INTEGER),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("1.2.3.4"), DataTypes.IP, "ip"),
+                        new TestCaseSupplier.TypedData(24, DataTypes.INTEGER, "prefixLengthV4")
+                    ),
+                    "IpPrefixOnlyV4Evaluator[ip=Attribute[channel=0], prefixLengthV4=Attribute[channel=1]]",
+                    DataTypes.IP,
+                    equalTo(EsqlDataTypeConverter.stringToIP("1.2.3.0"))
+                )
+            ),
+
+            // V6
+            new TestCaseSupplier(
+                List.of(DataTypes.IP, DataTypes.INTEGER),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("::ff"), DataTypes.IP, "ip"),
+                        new TestCaseSupplier.TypedData(24, DataTypes.INTEGER, "prefixLengthV4"),
+                        new TestCaseSupplier.TypedData(127, DataTypes.INTEGER, "prefixLengthV6")
+                    ),
+                    "IpPrefixEvaluator[ip=Attribute[channel=0], prefixLengthV4=Attribute[channel=1], prefixLengthV6=Attribute[channel=2]]",
+                    DataTypes.IP,
+                    equalTo(EsqlDataTypeConverter.stringToIP("::fe"))
+                )
+            ),
+            new TestCaseSupplier(
+                List.of(DataTypes.IP, DataTypes.INTEGER),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("::ff"), DataTypes.IP, "ip"),
+                        new TestCaseSupplier.TypedData(24, DataTypes.INTEGER, "prefixLengthV4")
+                    ),
+                    "IpPrefixOnlyV4Evaluator[ip=Attribute[channel=0], prefixLengthV4=Attribute[channel=1]]",
+                    DataTypes.IP,
+                    equalTo(EsqlDataTypeConverter.stringToIP("::ff"))
+                )
+            )
+        );
```

Comment:
It really would be nice to have some random examples just out of paranoia.

You could do something like:
```
() -> {
  INetAddress v4 = randomIp(true);
  int prefix = between(0, 32);
  InetAdderess expected = INetAddresses.parseIpRangeFromCidr(INetADdresses.toCidrString(v4, prefix)).lowerBound();
  return new TestCaseSupplier.TestCase(List..., equalTo(EsqlDataTypeConverter.stringToIP(INetADdresses.toAddrString(expected))
  return new TestCaseSupplier.TestCase(List.of
}
```

Or something. I can't write code without an ide any more, but something like this would go about the masking in another way so we get a little extra proof what we're doing is right.